src/main/java/org/cactoos/list/ImmutableListIterator.java

```diff
@@ -30,14 +30,14 @@
 import org.cactoos.scalar.Unchecked;
 
 /**
- * Iterator of the list that doesn't allow mutations.
+ * List Iterator that doesn't allow mutations.
  *
  * <p>There is no thread-safety guarantee.
  *
  * @param <T> Items type
  * @since 0.35
  */
-public final class ListIteratorOf<T> implements ListIterator<T> {
+public final class ImmutableListIterator<T> implements ListIterator<T> {
```

Comment:
@fanifieiev what about `ListIteratorOf`? I think it is still needed, it's better to introduce a new class for `ImmutableListIterator` (that only wraps a `ListIterator`, see my other comment below in this class). And Also add a todo to `ListIteratorOf` to make it not immutable.

Reply:
@victornoel, refactored in commits [4a6b4f993cdf92954c93fd260062e9396a35c7f9](https://github.com/yegor256/cactoos/pull/1227/commits/4a6b4f993cdf92954c93fd260062e9396a35c7f9) and [fd65977847f283da313a68afd02b6eec91e54719](https://github.com/yegor256/cactoos/pull/1227/commits/fd65977847f283da313a68afd02b6eec91e54719)